pyfakefs/fake_filesystem.py

```diff
@@ -1245,6 +1250,49 @@ def CreateLink(self, file_path, link_target):
     return self.CreateFile(resolved_file_path, st_mode=stat.S_IFLNK | PERM_DEF,
                            contents=link_target)
 
+  def CreateHardLink(self, old_path, new_path):
+    """Create a hard link at new_path, pointing at old_path.
+
+    Args:
+      old_path: an existing link to the target file
+      new_path: the destination path to create a new link at
+
+    Returns:
+      the FakeFile object referred to by old_path
+
+    Raises:
+      OSError:  if something already exists at new_path
+      OSError:  if the parent directory doesn't exist
+    """
+    new_path_normalized = self.NormalizePath(new_path)
+    if self.Exists(new_path_normalized):
+      raise IOError(errno.EEXIST,
+                    'File already exists in fake filesystem',
+                    new_path)
+
+    new_parent_directory, new_basename = self.SplitPath(new_path_normalized)
+    if not new_parent_directory:
+      new_parent_directory = self.cwd
+
+    if not self.Exists(new_parent_directory):
+      raise OSError(errno.ENOENT, 'No such fake directory',
+                    new_parent_directory)
+
+    # Retrieve the target file
+    try:
+      old_file = self.GetObject(old_path)
+    except:
+      raise OSError(errno.ENOENT,
+                    'No such file or directory in fake filesystem',
+                    old_path)
+
+    old_file.st_nlink += 1
+
+    # abuse the name field to control the filename of the newly created link
+    old_file.name = new_basename
+    self.AddObject(new_parent_directory, old_file)
```

Comment:
This will currently also increase the file system size by the linked file size, as st_nlink is not taken into account in AddEntry(). I may fix this after the PR is merged back, as I added the file system size stuff only recently and could do some minor code cleanup there anyway.